src/base/message.rs

```diff
@@ -439,6 +438,20 @@ impl<Octs: Octets + ?Sized> Message<Octs> {
         }
     }
 
+    /// Returns whether this is the answer to some other message.
+    ///
+    /// The method checks whether the ID fields of the headers are the same,
+    /// whether the QR flag is set in this message. It does NOT check whether
+    /// the questions are the same which is useful in case of responses such
+    /// as subsequent RFC 5936 AXFR responses which are not required to copy
+    /// the question from the request to the response.
+    pub fn is_answer_header<Other: Octets + ?Sized>(
```

Comment:
I think this is wrong. Low level transports check the ID, no need to check it at other places. In addition this seems only need for zone transfers. We should not put this in Message. It clutters the interface.

Reply:
The reason for this was that the `XfrResponseProcessor` has no direct link to client transports, anything that can provide a `Message` can use it.

Reply:
If somebody wants to apply XFR processing to a random collection of responses why check the ID? It needlessly complicates the interface. In any case, even if you want to check the ID in XfrResponseProcessor, just check it there. Do not add very specialized functions to Message.

Reply:
If someone accidentally processes unrelated messages it would help catch that problem.

I tend to prefer not making assumptions in library code about inputs but I appreciate there is a balance that can be found between defence-in-depth and defence-by-design, assuming that the API is constrained enough that the expected usage is known/guided in advance.

I didn't consider splitting an existing function to be adding very specialized behaviour.

I'll reconsider the code in question  in light of your comments.

Reply:
I removed the extra method as you requested.